tests/gpflow/experimental/check_shapes/test_errors.py

```diff
@@ -27,7 +27,7 @@ def context_func() -> None:
 def test_argument_reference_error() -> None:
     arg_map = MagicMock()
     arg_ref = MagicMock()
-    arg_ref.__str__.return_value = "my_argument_reference"
+    arg_ref.__str__.return_value = "my_argument_reference"  # type: ignore  # mypy confused by mock.
```

Comment:
I think this should work:

```suggestion
    arg_ref.__str__ = MagicMock(return_value = "my_argument_reference")
```

Reply:
> I think this should work:

Well, it needs to be:
```python
    arg_ref = MagicMock(__str__=MagicMock(return_value="my_argument_reference"))
```

But done.